src/main/java/io/supertokens/useridmapping/UserIdMapping.java

```diff
@@ -17,20 +17,73 @@
 package io.supertokens.useridmapping;
 
 import io.supertokens.Main;
+import io.supertokens.authRecipe.AuthRecipe;
 import io.supertokens.pluginInterface.authRecipe.AuthRecipeStorage;
+import io.supertokens.pluginInterface.emailverification.EmailVerificationStorage;
 import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.session.SessionStorage;
 import io.supertokens.pluginInterface.useridmapping.UserIdMappingStorage;
 import io.supertokens.pluginInterface.useridmapping.exception.UnknownSuperTokensUserIdException;
 import io.supertokens.pluginInterface.useridmapping.exception.UserIdMappingAlreadyExistsException;
+import io.supertokens.pluginInterface.usermetadata.UserMetadataStorage;
+import io.supertokens.pluginInterface.userroles.UserRolesStorage;
 import io.supertokens.storageLayer.StorageLayer;
+import io.supertokens.userroles.UserRoles;
 
 import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.HashMap;
 
 public class UserIdMapping {
 
     public static void createUserIdMapping(Main main, String superTokensUserId, String externalUserId,
             String externalUserIdInfo)
             throws UnknownSuperTokensUserIdException, UserIdMappingAlreadyExistsException, StorageQueryException {
+        createUserIdMapping(main, superTokensUserId, externalUserId, externalUserIdInfo, false);
+    }
+
+    public static void createUserIdMapping(Main main, String superTokensUserId, String externalUserId,
+            String externalUserIdInfo, boolean force)
+            throws UnknownSuperTokensUserIdException, UserIdMappingAlreadyExistsException, StorageQueryException {
+        // if a userIdMapping is created with force, then we skip the following checks
+        if (!force) {
+            // check that none of the non-auth recipes are using the superTokensUserId
+            {
+
+                if (StorageLayer.getStorage(main).isUserIdBeingUsedInNonAuthRecipe(SessionStorage.class.getName(),
+                        superTokensUserId)) {
+                    throw new IllegalStateException("SuperTokens Id is already in use in Session recipe");
```

Comment:
instead of illegalstate, you want to throw a bad request exception or something that is mapped to a 400 error